.github/workflows/auto-update-gemfiles.yml

```diff
@@ -0,0 +1,169 @@
+name: Auto-update Gemfiles
+
+# This action cannot be skipped altogether because it is a mandatory status check.
+# Instead we conditionally skip it at job level, instead of workflow level.
+on:
+  schedule:
+    - cron: '0 0 * * 0' # Every Sunday at midnight
+  workflow_dispatch:
+
+# Ensure obsolete job is cancelled if another commit is pushed to the same branch.
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  build:
+    runs-on: ubuntu-22.04
+    permissions:
+        actions: read # read secrets
+        contents: write
+    strategy:
+        fail-fast: false
+        matrix:
+            engine:
+              # ADD NEW RUBIES HERE
+              - name: ruby
+                version: '3.4'
+              - name: ruby
+                version: '3.3'
+              - name: ruby
+                version: '3.2'
+              - name: ruby
+                version: '3.1'
+              - name: ruby
+                version: '3.0'
+              - name: ruby
+                version: '2.7'
+              - name: ruby
+                version: '2.6'
+              - name: ruby
+                version: '2.5'
+              - name: jruby
+                version: '9.4'
+              - name: jruby
+                version: '9.3'
+              - name: jruby
+                version: '9.2'
+    container:
+      image: "ghcr.io/datadog/images-rb/engines/${{ matrix.engine.name }}:${{ matrix.engine.version }}"
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Output Ruby version
+        run: ruby -v
+
+      - name: Bundle
+        run: bundle install
+
+      - name: Update latest
+        run: bundle exec rake edge:latest
+
+      - name: Upload artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: 'datadog-gem-${{ matrix.engine.name }}-${{ matrix.engine.version }}-gha${{ github.run_id }}'
+          path: gemfiles/${{ matrix.engine.name }}_${{ matrix.engine.version }}_*
+          retention-days: 3
+
+  aggregate:
+    needs: build
+    runs-on: ubuntu-latest
+    permissions:
+      actions: read # read secrets
+      pull-requests: write
+      contents: write
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Download artifacts for Ruby 2.5
```

Comment:
Refactor with `merge-multiple` and `pattern` 

https://github.com/actions/download-artifact/tree/v4/?tab=readme-ov-file#download-multiple-filtered-artifacts-to-the-same-directory

Reply:
Nice!